metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -557,16 +557,17 @@ def _find_measure_recipe(
         logger.info(f"Found {len(node_to_evaluation)} candidate measure nodes.")
 
         if len(node_to_evaluation) > 0:
-            cost_function = DefaultCostFunction()
-
-            node_with_lowest_cost = min(node_to_evaluation, key=cost_function.calculate_cost)
-            evaluation = node_to_evaluation[node_with_lowest_cost]
+            # All source nodes cost the same. Find evaluation with lowest number of joins.
+            node_with_lowest_cost_plan = min(
+                node_to_evaluation, key=lambda node: len(node_to_evaluation[node].join_recipes)
+            )
+            evaluation = node_to_evaluation[node_with_lowest_cost_plan]
             logger.info(
-                "Lowest cost node is:\n"
+                "Lowest cost plan is:\n"
                 + pformat_big_objects(
-                    lowest_cost_node=dataflow_dag_as_text(node_with_lowest_cost),
+                    node=dataflow_dag_as_text(node_with_lowest_cost_plan),
                     evaluation=evaluation,
-                    cost=cost_function.calculate_cost(node_with_lowest_cost),
+                    joins=len(node_to_evaluation[node_with_lowest_cost_plan].join_recipes),
```

Comment:
Nice find!

We have Yet Another Inscrutable Visitor doing this complex cost calculation, but it can only go from leaf node to root node, not the other way around. Since measure nodes are currently ONLY source nodes the cost comparison is pointless.

Using minimum join count is probably the right answer here. Later on it'll matter more what the user requests, because eventually the measure nodes could be on the right. In theory, this kind of graph walk cost computation will be useful then, but my worry about this block of logic was always that it the results might diverge as the join layout changes. However, since we have now committed to always picking the shortest join paths I think this gets us closer to where we need to be.